README.md

```diff
@@ -1 +1,412 @@
+
+```objc
+// correct: using an initializer to give the properties initial values
+
+NSSortDescriptor *sortByNameAsc = [[NSSortDescriptor alloc] initWithKey:@"name"
+                                                              ascending:YES];
+```
+
+![](https://curriculum-content.s3.amazonaws.com/ios-intro-to-objects-unit/readonly_init.png)
+
+The usefulness of providing an initializer allows the user of the class to be assured that the instance will be set up holistically enough to perform its methods without unexpected behavior. It also allows us, the developer, to document the set up for the class.
+
+### The Designated Initializer
+
+A "designated initializer" is the initializer method which provides the *most thorough coverage of that class's data.* It doesn't necessarily provide *complete* coverage, just the *most* coverage of any of the `init...` methods. Take for example, the `initWithObjects:` method available to `NSMutableArray`.
+
+**Note:** *The* `initWithObjects:` *method is actually defined on the* `NSArray` *class, but since* `NSMutableArray` *is a subclass of* `NSArray` *it can also perform the* `initWithObjects:` *method.*
+
+```objc
+// designated initializer
+
+NSMutableArray *instructors = [[NSMutableArray alloc] initWithObjects:@"Joe", @"Tim", @"Tom", @"Jim", @"Mark", nil];
+
+NSLog(@"%@", instructors);
+```
+This will print:
+
+```
+(
+    Joe,
+    Tim,
+    Tom,
+    Jim,
+    Mark
+)
+```
+
+The role of a designated initializer is that it returns an instance of the given class (`NSMutableArray` in this case) that's considered prepared enough for actual use. We could get to this state manually by using the default initializer `init` and then calling the `addObjectsFromArray:` method to insert the list of objects into the array:
+
+```objc
+// manual set up
+
+NSMutableArray *instructors = [[NSMutableArray alloc] init];
+[instructors addObjectsFromArray:@[@"Joe", @"Tim", @"Tom", @"Jim", @"Mark"]];
+
+NSLog(@"%@", instructors);
+```
+
+This will also print:
+
+```
+(
+    Joe,
+    Tim,
+    Tom,
+    Jim,
+    Mark
+)
+```
+It's likely that the `init` method actually calls the `initWithObjects:` method but with an empty list of objects to be added. At some point, a designated initializer **must** be called when creating a new instance; other initializers that are not defined as designated initializers **must** pass through a designated initializer. This is a way that the compiler helps to ensure that the instance will be functional at run time.
+
+**Note:** *While this is not strictly enforced by Objective-C itself, it is considered best practice—especially when writing Objective-C that will interface with anything written Swift which does hold strict rules about designated initializers.*
+
+### Declaring the Designated Initializer
+
+Since initializers *must* be public methods, they need to be declared in the `.h` header file. The rules for `init` family methods that are enforced by Apple via the compiler are that it:
```

Comment:
Can't think of anything offhand, but I know I've done it at least once. Either way, I would just put it like.. "Like other public methods, initializers need to be declared..."